### DISCUSSION TITLE
Possibility to localize delimiter in CSV-export

### DISCUSSION THREAD

**Discussion by luke-nehemedia** [craftcms/cms#14772] [Ideas] [answered]
I am based in Germany and here, Microsoft Excel will fail to open a CSV-file that uses `,` as the delimiter. In Germany, `;` is commonly used, probably because `,` is used for decimals. Apple Numbers somehow will open both variants without problems. However, most of my clients use Microsoft Excel.

Right now, the delimiter is set as a variable in [CsvResponseFormatter.php](src/web/CsvResponseFormatter.php). 

Could you make this a localizable variable so one can change it to `;`?

**Comment by brandonkelly** [comment, marked as answer]
You can accomplish this by creating a `config/app.web.php` file, with the following:

```php
<?php

use craft\helpers\App;
use craft\web\CsvResponseFormatter;
use craft\web\Response;

return [
    'components' => [
        'response' => fn() => Craft::createObject(array_merge(
            App::webResponseConfig(),
            [
                'formatters' => [
                    Response::FORMAT_CSV => [
                        'class' => CsvResponseFormatter::class,
                        'delimiter' => ';',
                    ],
                ],
            ],
        )),
    ],
];
```
